rockchip: add FriendlyElec NanoPi R6C and fixes for the NanoPi R6S

antnyfls opened the pull request:
1- This add support for the FriendlyElec NanoPi R6C 
2 Fixes for the NanoPi R6S pending on the kernel mailing list'
https://lore.kernel.org/all/20240612205056.397204-1-seb-dev@mail.de

antnyfls committed:
uboot-rockchip: add FriendlyElec NanoPi R6C

1- The NanoPi R6C is a SBC by FriendlyElec based on the Rockchip RK3588s.
   It comes with 4GB or 8GB of RAM, a microSD card slot, optional 32GB eMMC
   storage, one M.2 M-Key connector, one RTL8211F 1GbE and one RTL8125
   2.5GbE Ethernet port, one USB 2.0 Type-A and one USB 3.0 Type-A port, a
   HDMI port, a 30-pin GPIO header as well as multiple buttons and LEDs.
2- Renamed 000-backport-upstream-dts-sync.patch -> 000-v2024.10-rc1-backport-upstream-dts-sync.patch
   to add the version when was applied upstream

Signed-off-by: Antonio Flores <antflores627@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/16275
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

antnyfls committed:
rockchip: add FriendlyElec NanoPi R6C

Hardware Spec
 -SoC: Rockchip RK3588S
       CPU: Quad-core ARM Cortex-A76(up to 2.4GHz) and quad-core Cortex-A55 CPU (up to 1.8GHz)
       GPU: Mali-G610 MP4, compatible with OpenGLES 1.1, 2.0, and 3.2, OpenCL up to 2.2 and Vulkan1.2
       VPU: 8K@60fps H.265 and VP9 decoder, 8K@30fps H.264 decoder, 4K@60fps AV1 decoder, 8K@30fps H.264 and H.265
       NPU: 6TOPs, supports INT4/INT8/INT16/FP16
 -RAM: 64-bit 4GB/8GB LPDDR4X at 2133MHz
 -Flash: 32GB/None eMMC, at HS400 mode
 -Ethernet: one Native Gigabit Ethernet, and one PCIe 2.5G Ethernet
 -USB: one USB 3.0 Type-A and one USB 2.0 Type-A
 -PCIe: one M.2 Key M connector with PCIe 2.1 x1
 -HDMI:
       compatible with HDMI2.1, HDMI2.0, and HDMI1.4 operation
       support up to 7680x4320@60Hz
       Support RGB/YUV(up to 10bit) format
 -microSD: support up to SDR104 mode
 -GPIO:
       30-pin 2.54mm header connector
       up to 1x SPI, 3x UARTs, 3x I2Cs, 2x SPDIFs, 1x I2Ss, 3x PWMs, 20x GPIOs
 -Debug: UART via 3-Pin 2.54mm header, or on-board USB-C to UART
 -LEDs: 4 x GPIO Controlled LED (SYS, WAN, LAN, LED1)
 -others:
        2 Pin 1.27/1.25mm RTC battery input connector for low power RTC IC HYM8563TS
        MASK button for eMMC update
        one user button
 -Power supply: USB-C, support PD, 5V/9V/12V/20V input
 -PCB: 8 Layer, 62x90x1.6mm
 -Ambient Operating Temperature: 0℃ to 70℃

Installation:

Uncompress the OpenWrt sysupgrade and write image to the SD card using dd (dd if=*.img of=/*)

eMMC Installation:

Boot from the SD card
Uncompress the OpenWrt sysupgrade image
fash to eMMC : dd if=*.img of=/dev/mmcblk1
sync
remove SD card
reboot

Signed-off-by: Antonio Flores <antflores627@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/16275
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

antnyfls committed:
rockchip: fixes and improve for NanoPi R6 series

Improvement and fixes for the nanopi r6 series, pending on the kernel mailing list
https://lore.kernel.org/all/20240612205056.397204-1-seb-dev@mail.de

Signed-off-by: Antonio Flores <antflores627@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/16275
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

odkrys commented:
I created an r6s image with this PR and tested it. I don't know exactly why, but it seems to have fixed the port swap issue. It seems like the pcie ethernets boot in order.

https://github.com/openwrt/openwrt/issues/16271

And the usb3.0 port is also working.

potus002 commented:
keep up the great work

blocktrron commented:
Looks good. For the U-Boot patches i assume they have been applied upstream? If so, can you add the version it is / will first be included in? You already did this for the target patch which is not pending.

This makes it easier to identify which patches to drop when updating the base version.

antnyfls commented:
> Looks good. For the U-Boot patches i assume they have been applied upstream? If so, can you add the version it is / will first be included in? You already did this for the target patch which is not pending.
> 
> This makes it easier to identify which patches to drop when updating the base version.

I did include the version on the new patch
`107-v2024.10-rc1-board-rockchip-Add-FriendlyElec-NanoPi-R6C.patch`
patch `001-backport-upstream-dts-sync.patch` is already applied, I only added the r6c DTS .

antnyfls committed:
uboot-rockchip: patches: add version when were applied upstream

This patch add a version to uboot patches to help identify in
futures updates when they were upstreamed.

Signed-off-by: Antonio Flores <antflores627@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/16275
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

antnyfls commented:
I added the version in all u boot-rockchip patches, non of them have it right now, in a new patch.

dick902 commented:
Tested with Nano-Pi R6C, everything ok for me.

gSpotx2f commented:
I tested with NanoPC R6S. It looks like the 2.5G ports issue (non-deterministic interface naming) is fixed. USB3 port works fine.
But there is a small issue with the [new LED names](https://lore.kernel.org/all/20240612205056.397204-4-seb-dev@mail.de/) in `/etc/board.d/01_leds`:

    --- a/target/linux/rockchip/armv8/base-files/etc/board.d/01_leds
    +++ b/target/linux/rockchip/armv8/base-files/etc/board.d/01_leds
    @@ -29,13 +29,13 @@ friendlyarm,nanopi-r5s)
            ucidef_set_led_netdev "lan2" "LAN2" "green:lan-2" "eth2"
            ;;
     friendlyarm,nanopi-r6c)
    -       ucidef_set_led_netdev "wan" "WAN" "wan_led" "eth0"
    -       ucidef_set_led_netdev "lan1" "LAN1" "lan1_led" "eth1"
    +       ucidef_set_led_netdev "wan" "WAN" "green:wan" "eth0"
    +       ucidef_set_led_netdev "lan1" "LAN1" "green:lan-1" "eth1"
            ;;
     friendlyarm,nanopi-r6s)
    -       ucidef_set_led_netdev "wan" "WAN" "wan_led" "eth1"
    -       ucidef_set_led_netdev "lan1" "LAN1" "lan1_led" "eth2"
    -       ucidef_set_led_netdev "lan2" "LAN2" "lan2_led" "eth0"
    +       ucidef_set_led_netdev "wan" "WAN" "green:wan" "eth1"
    +       ucidef_set_led_netdev "lan1" "LAN1" "green:lan-1" "eth2"
    +       ucidef_set_led_netdev "lan2" "LAN2" "green:lan-2" "eth0"
            ;;
     esac

antnyfls committed:
rockchip: rework LED configurations for the NanoPi R6C/R6S

This commits fixes the LED on the NanoPi R6 series
after changes in the DTS
https://lore.kernel.org/all/20240612205056.397204-4-seb-dev@mail.de
Reported by Github user: gSpotx2f

Signed-off-by: Antonio Flores <antflores627@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/16275
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

antnyfls commented:
> But there is a small issue with the [new LED names](https://lore.kernel.org/all/20240612205056.397204-4-seb-dev@mail.de/) in `/etc/board.d/01_leds`:

@gSpotx2f thank you for reporting, after reset my device I noticed the LED problem, Tested only the R6C.

bigsmile74 reviewed (approved):
Build the image today for the R6C using this PR. Wrote it to sdcard and it boots. I browsed through it a bit and it seems to work fine. Installed luci and that seems to work fine too. NICs seem in order. Bootlog was clean afaict.

I like it :smile:

hauke commented on target/linux/rockchip/patches-6.6/400-1-arm64-dts-rockchip-Add-common-definitions-for-NanoPi.patch:
```diff
@@ -0,0 +1,1573 @@
+From 93be10e4e180eeac9b6a57a4363e08c35bee3af9 Mon Sep 17 00:00:00 2001
+From: Sebastian Kropatsch <seb-dev@mail.de>
```
Is someone working on getting these patches accepted upstream?
I can not find them in linux-next.

antnyfls replied:
> Is someone working on getting these patches accepted upstream?
> I can not find them in linux-next.

have not been merged yet, they are being reviewed https://lore.kernel.org/all/?q=nanopi-r6

hauke replied:
The patches were send about 2.5 months ago. Do you know if a updated version will be send soon?

antnyfls replied:
I don't have that info, if is strictly a  requirement that the patches be in the kernel tree I can drop `rockchip: fixes and improve for NanoPi R6 series` and `rockchip: rework LED configurations for the NanoPi R6C/R6S`  for later if they get accepted. is OK that?

hauke replied:
The patches are not looking so bad, I applied your PR.

Please send a new PR when these patches are getting merged upstream.

hariskar commented:
I just got my nanopi R6S, installed OpenWrt 24.10.0 r28427-6df0e3d02a / LuCI openwrt-24.10 branch 25.014.55016~7046a1c but I have problem with the ports: If I assign eth2 to WAN interface (eth2 is preconfigured in WAN) it correspondents to the 2nd physical port whuch is LAN1. So I have to put the WAN ethernet cable to LAN1 port instead of the WAN port. Any solution?

aenertia commented:
Random port swaps are happening still for me with nanopc-t6 with 24.10.2 still. I have burned mac address's permanently with the ratool ( https://github.com/redchenjs/rtnicpg ) thinking it would solve the issue permanently - but obviously the dtb enumeration is still being referenced rather than MAC somewhere during init.
